### PR TITLE
Feat: multi file question

### PR DESCRIPTION
**Pull request by winged**
### Re-merge the feature from #1809

Change the model to allow multiple files per answer to be stored. Migrate
the existing questions to the new question type, and migrate the existing
answers to the new schema.

Updated the GraphQL schema to allow saving and updating multiple files
on an answer. The answer's value is a list of dicts, each containing
a file name, and optionally an ID. When you're creating new files,
add an entry without ID to the value. Caluma will then save the file,
and return it back WITH and ID.

BREAKING CHANGE: This renames the question type constant for file questions,
and changes the semantics of the answer value for file questions as well:
It is now a list of dicts instead of a single string. The response type
for querying file(s) answers now also is a list instead of a single dict.

For details / RFC see https://github.com/projectcaluma/caluma/issues/1780

### Dependencies:

* https://github.com/projectcaluma/ember-caluma/pull/2040 - should be merged simoultaneously

Supersedes #1813 which is incomplete

### CODE REVIEW

File: caluma/caluma_form/domain_logic.py [hunk 1]
```diff
@@ -91,35 +142,35 @@ def post_save(answer: models.Answer) -> models.Answer:
...
             answer.unlink_unused_rows(docs_to_keep=documents)
 
-        if (
-            answer.question.type == models.Question.TYPE_FILE
-            and answer.file.name is not validated_data["file"]
-        ):
-            answer.file.delete()
-            validated_data = __class__.set_file(validated_data)
+        if answer.question.type == models.Question.TYPE_FILES:
+            cls.update_answer_files(answer, validated_data.pop("files", None))
```
Comment (StephanH90) [inline caluma/caluma_form/domain_logic.py:173]: Similarly here, if on python >= 3.10 could use a `match`.

File: caluma/caluma_form/structure.py [hunk 2]
```diff
@@ -91,8 +91,8 @@ def value(self):
                 for row in self.children()
             ]
 
-        elif self.question.type == Question.TYPE_FILE:
-            return self.answer.file
+        elif self.question.type == Question.TYPE_FILES:
+            return [f.name for f in self.answer.files.all()]
```
Comment (StephanH90) [inline caluma/caluma_form/structure.py:95]: Now we return the `name` of the file while before we returned the entire `file`. Is that the same in this context or is that an intended change?

File: caluma/caluma_form/domain_logic.py [hunk 3]
```diff
@@ -42,13 +43,63 @@ def get_new_answer(cls, data, user, answer):
...
+        if not files:
+            files = []
+
+        updated = []
+
+        for file_ in files:
+            file_id = extract_global_id(file_["id"]) if "id" in file_ else None
+            file_name = file_["name"]
+            file_model = (
+                answer.files.filter(pk=file_id).first()
```
[the remark this answers is not in the document]
Comment (winged) [reply caluma/caluma_form/domain_logic.py:78]: Nope - different behaviour. I don't wanna crash/fail when a file is gone (might have be a parallel edit, whole other story)

File: caluma/caluma_form/domain_logic.py [hunk 4]
```diff
@@ -91,35 +142,35 @@ def post_save(answer: models.Answer) -> models.Answer:
...
+        cls, validated_data: dict, user: Optional[BaseUser] = None
+    ) -> models.Answer:
 
         if validated_data["question"].type == models.Question.TYPE_TABLE:
             documents = validated_data.pop("documents")
 
+        files = validated_data.pop("files", None)
         answer = BaseLogic.create(models.Answer, validated_data, user)
 
+        if validated_data["question"].type == models.Question.TYPE_FILES:
```
[the remark this answers is not in the document]
Comment (StephanH90) [reply caluma/caluma_form/domain_logic.py:157]: I think while we are here we might as well make the code tidy but that is up to you (I didn't run this code):

```python
files = validated_data.pop("files", None)
answer = BaseLogic.create(models.Answer, validated_data, user)

if answer.question.type == models.Question.TYPE_TABLE:
    answer.create_answer_documents(validated_data.pop("documents"))

if answer.question.type == models.Question.TYPE_FILES:
    cls.update_answer_files(answer, files)
    
return answer
```

**Review by StephanH90 — APPROVED** [review]

**Pull request merged**